pallets/parachain-staking/src/set.rs

```diff
@@ -191,54 +219,71 @@ impl<T: Ord> OrderedSet<T> {
 		self.0.is_empty()
 	}
 
-	/// Convert the set to a vector.
-	pub fn into_vec(self) -> Vec<T> {
+	/// Convert the set to a bounded vector.
+	pub fn into_bounded_vec(self) -> BoundedVec<T, S> {
 		self.0
 	}
 
-	/// Sort from greatest to lowest
+	/// Sorts from greatest to lowest.
+	///
+	/// NOTE: BoundedVec does not implement DerefMut because it would allow for
+	/// unchecked extension of the inner vector. Thus, we have to work with a
+	/// clone unfortunately.
 	pub fn sort_greatest_to_lowest(&mut self) {
-		self.0.sort_by(|a, b| b.cmp(a));
+		let mut sorted_v = self.0.clone().into_inner();
+		sorted_v.sort_by(|a, b| b.cmp(a));
+		self.0 = sorted_v.try_into().expect("Did not extend size of bounded vec");
 	}
 }
 
-impl<T: Ord> From<Vec<T>> for OrderedSet<T> {
-	fn from(v: Vec<T>) -> Self {
-		Self::from(v)
+#[cfg(feature = "std")]
+impl<T, S> fmt::Debug for OrderedSet<T, S>
+where
+	T: fmt::Debug,
+	S: Get<u32>,
+{
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.debug_tuple("OrderedSet").field(&self.0).finish()
 	}
 }
```

Comment:
We need that only in std?

Reply:
No but `self.0` (the bounded vec) only implements `Debug` in `std`. The compiler was happy with restricting to `std`. Should I change the `Debug` implementation to be working in all features instead?

Reply:
I was just wondering. But no that makes sense. leave it like it is I would say.